File: plugins/remote/wastool/bin/python/was-cluster-start.py
#!/usr/bin/python
import ConfigParser
import os
import sys
import utils
import re

if len(sys.argv) < 2:
	progName = os.path.basename(__file__)
	print("ERROR:use as %s config-name instance-name\n" % progName)
	sys.exit(1)

homepath = os.path.split(os.path.realpath(__file__))[0]
homepath = os.path.realpath("%s/.." % homepath)
os.chdir(homepath)

config = ConfigParser.ConfigParser()
config.readfp(open('conf/was.ini'))

#test section name, sequence: mainname.insname -> mainname.insprefix -> mainname
mainname = sys.argv[1]
insname = None
if len(sys.argv) == 3:
        insname = sys.argv[2]

confname = mainname
if insname is not None:
        insprefix = re.sub(r'\d*$', '', insname)
        sections = config.sections()
        if '%s.%s' % (mainname, insname) in sections:
                confname = '%s.%s' % (mainname, insname)
        elif '%s.%s' % (mainname, insprefix) in sections:
                confname = '%s.%s' % (mainname, insprefix)
else:
        print("ERROR:use as was-cluster-start.py config-name instance-name\n")
        sys.exit(1)


wasprofile = config.get(confname,'was_profile')
was_user = config.get(confname,'was_user')
was_pwd = config.get(confname,'was_pwd')
appname= config.get(confname,'appname')

appname = appname.replace(' ', '')
appname = appname.replace("\t", '')
appnames = appname.split(',')

for appname in appnames:
	precheckurl = None
	try:
		precheckurl = config.get(confname, '%s.precheckurl' % appname)
	except:
		pass
	utils.checkUtilUrlAvaliable(precheckurl)

cmd = '%s/bin/wsadmin.sh  -lang jython -user %s -password %s -f %s/bin/was-cluster-poststart.py' % (wasprofile,was_user,was_pwd,homepath)
os.environ['TS_WASDEPLOYTOOL_HOME'] = homepath
os.environ['TS_WASCONF_NAME'] = confname
os.environ['TS_WASINS_NAME'] = insname
os.system(cmd)


for appname in appnames:
	checkurl = None
	try:
		checkurl = config.get(confname, '%s.checkurl' % appname)
	except:
		pass
	utils.checkUtilUrlAvaliable(checkurl)

File: plugins/remote/wastool/bin/was-cluster-start.pl
#!/usr/bin/perl
use strict;
use FindBin;
use lib "$FindBin::Bin";
use Utils;

use POSIX qw(uname);

my @uname    = uname();
my $ostype   = $uname[0];
my $shellExt = 'sh';
my $nullDev  = '/dev/null';

if ( $ostype =~ /Windows/i ) {
    $ostype   = 'windows';
    $shellExt = 'bat';
    $nullDev  = 'NUL';
}
else {
    $ostype   = 'unix';
    $shellExt = 'sh';
    $nullDev  = '/dev/null';
}

use CommonConfig;
use Cwd 'abs_path';
my $homePath = $FindBin::Bin;
$homePath = abs_path("$homePath/..");

#$ENV{LANG} = 'utf-8';

if ( scalar(@ARGV) != 2 ) {
    my $progName = $FindBin::Script;
    print("ERROR: Use as $progName config-name instance-name\n");
    exit(1);
}

chdir($homePath);
my $configName = $ARGV[0];
my $insName    = $ARGV[1];
my $insPrefix  = $insName;
$insPrefix =~ s/\d*$//;

my $sectionConfig;
my $config = CommonConfig->new( "$homePath/conf", "wastool.ini" );
$sectionConfig = $config->getConfig("$configName.$insName");
my $confName = "$configName.$insName";
if ( not defined($sectionConfig) ) {
    $sectionConfig = $config->getConfig("$configName.$insPrefix");
    $confName      = "$configName.$insPrefix";
}
if ( not defined($sectionConfig) ) {
    $confName      = $configName;
    $sectionConfig = $config->getConfig("$configName");
}
my $wasprofile = $sectionConfig->{"was_profile"};
my $wasUser    = $sectionConfig->{"was_user"};
my $wasPwd     = $sectionConfig->{"was_pwd"};

my $umask = $sectionConfig->{"umask"};
if ( defined($umask) and $umask ne '' ) {
    umask($umask);
}

my $startTimeout = $sectionConfig->{'start_timeout'};
if ( not defined($startTimeout) or $startTimeout eq '' ) {
    $startTimeout = 300;
}

my $timeout = int($startTimeout);

my $appName = $sectionConfig->{"appname"};
$appName =~ s/\s*//g;
my @appNames = split( ",", $appName );

foreach $appName (@appNames) {
    my $checkUrl = $sectionConfig->{"$appName.precheckurl"};
    if ( defined($checkUrl) and $checkUrl ne '' ) {
        if ( not Utils::CheckUrlAvailable( $checkUrl, "GET", $timeout ) ) {
            print("ERROR: Pre-check url $checkUrl is not available in $timeout s, starting halt.\n");
            exit(1);
        }
    }
}

my $cmd = "$wasprofile/bin/wsadmin.$shellExt  -lang jython -user $wasUser -password $wasPwd -f $homePath/bin/was-cluster-poststart.py";
$cmd = "\"$wasprofile/bin/wsadmin.$shellExt\"  -lang jython -user $wasUser -password $wasPwd -f \"$homePath/bin/was-cluster-poststart.py\""
    if ( $ostype eq 'windows' );
if ( not defined($wasPwd) or $wasPwd eq '' ) {
    $cmd = "$wasprofile/bin/wsadmin.sh  -lang jython -f $homePath/bin/was-cluster-poststart.py";
    $cmd = "\"$wasprofile/bin/wsadmin.sh\"  -lang jython -f \"$homePath/bin/was-cluster-poststart.py\"" if ( $ostype eq 'windows' );
}
$ENV{'TS_WASDEPLOYTOOL_HOME'} = $homePath;
$ENV{'TS_WASCONF_NAME'}       = $confName;
$ENV{'TS_WASINS_NAME'}        = $insName;

#system($cmd);
Utils::execCmd($cmd);

foreach $appName (@appNames) {
    my $checkUrl = $sectionConfig->{"$appName.checkurl"};
    if ( defined($checkUrl) and $checkUrl ne '' ) {
        if ( not Utils::CheckUrlAvailable( $checkUrl, "GET", $timeout ) ) {
            print("ERROR: Check url $checkUrl is not available in $timeout s, starting halt.\n");
            exit(1);
        }
    }
}
